engine/src/witness/btc/smart_contract.rs

```diff
@@ -220,13 +226,12 @@ mod tests {
 	fn test_extract_contract_call_from_tx() {
```

Comment:
Would be nice to have a test to make sure the encode does not change. Like we have done in other places.